nntrainer/src/bn_layer.cpp

```diff
@@ -83,37 +94,34 @@ sharedConstTensor BatchNormalizationLayer::forwarding(sharedConstTensor in) {
   Tensor &gamma = paramsAt(static_cast<int>(BNParams::gamma)).weight;
   Tensor &beta = paramsAt(static_cast<int>(BNParams::beta)).weight;
 
-  if (trainable) {
-    Tensor deviation;
-    input = *in;
-
-    ///< current mu */
-    Tensor cmu;
-
-    cmu = input.average(0);
+  Tensor deviation;
 
+  input = *in;
+  /// @todo change trainable #524
+  if (trainable) {
+    Tensor cmu = input.average(axes_to_reduce);
     deviation = input.subtract(cmu);
 
-    this->cvar = deviation.chain()
-                   .multiply_i(deviation)
-                   .sum(0)
-                   .multiply_i(1.0f / input_dim.batch())
-                   .add_i(epsilon)
-                   .run();
+    cvar = deviation.multiply(deviation).average(axes_to_reduce);
+    cvar.add_i(epsilon);
 
-    /// @todo replace momentum paramter
+    /// @todo replace momentum parameter to prop
     float momentum = 0.9;
     mu.multiply_i(momentum);
     mu.add_i(cmu, 1 - momentum);
     var.multiply_i(momentum);
     var.add_i(cvar, 1 - momentum);
 
+    std::cerr << cvar;
```

Comment:
is this intentional? if you want to keep logging messages for debugging in patches, let's stick to `ml_logd`.

Reply:
No it's mistake. sry about that.